sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -490,7 +508,10 @@ def _add_step(self, step_kind, step_label, transform_node, side_tags=()):
     self.job.proto.steps.append(step.proto)
     step.add_property(PropertyNames.USER_NAME, step_label)
     # Cache the node/step association for the main output of the transform node.
-    self._cache.cache_output(transform_node, None, step)
+    output_tag = (
```

Comment:
Need a TODO here about the main output of external transforms being ambiguous. Or, could it be the only element of set(ransform_node.outputs.keys()) - set(side_tags)?